assets/src/scripts/inc/history.js

```diff
@@ -0,0 +1,282 @@
+							revision.active = false;
+						}
+
+						return (
+							<PageHistoryListItem key={ revision.id } id={ revision.id } action={ revision.action } author={ revision.author } date={ revision.date } content={ revision.content } active={ revision.active } actions={ actions } />
+						);
+					}) }
+				</ul>
+			</div>
+		);
+
+	},
+
+});
+
+var PageHistoryListItem = React.createClass({
+
+	getDefaultProps: function() {
+		return {
+			id:      0,
+			active:  false,
+			author:  '',
+			action:  'update',
+			date:    '',
+			content: '',
+		};
+	},
+
+	render: function() {
+
+		var verb       = 'create' === this.props.action ? 'created' : 'updated';
+		var classNames = [ 'PageHistory_List_Item' ];
+
+		if ( this.props.active ) {
+			classNames.push( 'PageHistory_List_Item-Active' );
+		}
+
+		return (
+			<li className={ classNames.join( ' ' ) } onClick={ this.handleSelectRevision }>
+				{ this.props.author } { verb } { this.props.date }
+			</li>
+		);
+	},
+
+	handleSelectRevision: function(e) {
+		this.props.actions.selectRevision( this.props );
+	},
+
+});
+
+var PageHistory = React.createClass({
+
+	getInitialState: function() {
+		return {
+			revisions: this.props.revisions,
+			diff: { a: null, b: null },
+		};
+	},
+
+	render: function() {
+
+		const actions = {
+			selectRevision: this.selectRevision,
+		};
+
+		return (
+			<div>
+				<PageHistoryDiff diff_a={ this.state.diff.a } diff_b={ this.state.diff.b } />
+				<PageHistoryList revisions={ this.state.revisions } actions={ actions } />
+			</div>
+		);
+	},
+
+	/**
+	 * Get the revision to compare against.
+	 * This should always be the next most recent revision.
+	 */
+	getNextRevision: function( revision ) {
+
+		var currentIndex, revision_b;
+
+		if ( ! revision ) {
+			return null;
+		}
+
+		this.props.revisions.forEach( function( _revision, i ) {
+			if ( _revision.id === revision.id )  {
+				currentIndex = i;
+			}
+		} );
+
+		if ( 'undefined' === typeof currentIndex ) {
+			return null;
+		}
+
+		if ( currentIndex < this.props.revisions.length - 1 ) {
+			revision_b = this.props.revisions[ currentIndex + 1 ];
+		} else {
+			revision_b = this.props.revisions[ this.props.revisions.length - 1 ];
+		}
+
+		return revision_b;
+
+	},
+
+	selectRevision: function( revision ) {
+
+		// If trying to select again, clear.
+		if ( this.state.diff.a && revision.id === this.state.diff.a.id ) {
+			revision = null;
+		}
+
+		// Update active state for each revision.
+		var newRevisions = this.state.revisions.map( function( _revision ) {
+
+			if ( revision && revision.id === _revision.id ) {
+				_revision.active = true;
+			} else {
+				_revision.active = false;
+			}
+
+			return _revision;
+
+		} );
+
+		this.setState({
+			revisions: newRevisions,
+			diff:      { a: revision, b: this.getNextRevision( revision ) },
+		});
+
+		// Defer until actually set.
+		window.setTimeout( function() {
+			this.toggleContainerClass();
+		}.bind( this ) );
```

Comment:
ES6 closures automatically bind, so can be `window.setTimeout( () => this.toggleContainerClass() );`